source/dub/recipe/packagerecipe.d

```diff
@@ -524,15 +526,18 @@ struct BuildSettingsTemplate {
 
  		// collect source files
 		dst.addSourceFiles(collectFiles(sourcePaths, "*.d"));
+		dst.addSourceFiles(collectFiles(cSourcePaths, "*.{c,i}"));
 		auto sourceFiles = dst.sourceFiles.sort();
 
  		// collect import files and remove sources
 		import std.algorithm : copy, setDifference;
 
-		auto importFiles = collectFiles(importPaths, "*.{d,di}").sort();
+		auto importFiles =
+            chain(collectFiles(importPaths, "*.{d,di}"), collectFiles(cImportPaths, "*.h"))
+            .sort();
```

Comment:
mixed tabs and spaces + put the `.array` before `.sort()` because it needs memory to operate on (also makes the .release.array changes below obsolete)

Reply:
Reworked my code.